### ISSUE TITLE
[BUG] serivce-ca crash looping

### ISSUE THREAD

**Issue by cooktheryan** [redhat-et/microshift#356]
#### What happened:
```
openshift-service-ca            service-ca-84b44986cb-wsqxs           0/1     CrashLoopBackOff   5          16m
```

#### What you expected to happen:
Service ca stays online

#### How to reproduce it (as minimally and precisely as possible):

1. Deploy from install.sh or build from main
2. kubectl get po -wA

#### Environment:
- Microshift version (use `microshift version`): All
- Hardware configuration: amd64
- OS (e.g: `cat /etc/os-release`): 8.4 rhel and fedora
- Kernel (e.g. `uname -a`): 4.18.0-305.el8.x86_64
- Others:

**Comment by fzdarsky** [comment]
Unfortunately, it's still an issue.

A workaround is to delete the crash-looping service-ca pod and let it restart.

**Comment by fzdarsky** [comment]
The reason for service-ca crash-looping is that it gets assigned an IP address from the wrong subnet and therefore can't connect which makes the process segfault:
```
[vagrant@fedora ~]$ oc logs pod/service-ca-7bffb6f6bf-sbnxl -n openshift-service-ca
W1221 13:18:36.329434       1 cmd.go:204] Using insecure, self-signed certificates
I1221 13:18:36.832263       1 observer_polling.go:159] Starting file observer
W1221 13:18:39.963082       1 builder.go:209] unable to get owner reference (falling back to namespace): Get "https://10.43.0.1:443/api/v1/namespaces/openshift-service-ca/pods": dial tcp 10.43.0.1:443: connect: no route to host
I1221 13:18:39.963316       1 builder.go:240] service-ca-controller version v3.11.0-357-gb66c450-
F1221 13:19:10.812187       1 cmd.go:129] unable to load configmap based request-header-client-ca-file: Get "https://10.43.0.1:443/api/v1/namespaces/kube-system/configmaps/extension-apiserver-authentication": dial tcp 10.43.0.1:443: connect: no route to host
goroutine 1 [running]:
k8s.io/klog/v2.stacks(0xc000012001, 0xc000578000, 0x103, 0x205)
	/go/src/github.com/openshift/service-ca-operator/vendor/k8s.io/klog/v2/klog.go:1021 +0xb9
k8s.io/klog/v2.(*loggingT).output(0x3835c20, 0xc000000003, 0x0, 0x0, 0xc0007d35e0, 0x2d8beaa, 0x6, 0x81, 0x414600)
	/go/src/github.com/openshift/service-ca-operator/vendor/k8s.io/klo

**Comment by fzdarsky** [comment]
A workaround is to delete both pods. They receive a correct IP after restart.

**Comment by fzdarsky** [comment]
Fixed in #519 and #525.
